File: module-1/18_Review/lecture-student/java/src/main/java/com/techelevator/search/SearchDomain.java
package com.techelevator.search;

import java.nio.file.DirectoryStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

/**
 * The Search Domain is a list of files to be read and indexed by the Search Engine.
 *
 * Folders are NOT recursively searched.
 */
public class SearchDomain {

	private final String folder;
	private final List<String> files;

	/**
	 * Create a Search Domain of a folder
	 *
	 * @param folder
	 * @throws SearchDomainException
	 */
	public SearchDomain(String folder) throws SearchDomainException {
		this.folder = folder;
		this.files = buildDomain();
	}

	public String getFolder() {
		return folder;
	}

	public List<String> getFiles() {
		return files;
	}

	@Override
	public String toString() {
		String result = "";
		for (String file : files) {
			result += file + "\n";
		}
		return result;
	}

	/**
	 * Folders are NOT recursively searched.
	 *
	 * @return
	 * @throws SearchDomainException
	 */
	private List<String> buildDomain() throws SearchDomainException {
		List<String> files = new ArrayList<>();

		Path dir = Path.of(folder);
		try (DirectoryStream<Path> allFilesInDir = Files.newDirectoryStream(dir)) {
			for (Path file: allFilesInDir) {
				files.add(file.toString());
			}
		} catch (Exception e) {
			throw new SearchDomainException(e.getMessage());
		}

		return files;
	}

}

File: module-1/18_Review/lecture-student/dotnet/SearchApplication/Search/SearchDomain.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace SearchApplication.Search
{
    /// <summary>
    /// The Search Domain is a list of files to be read and indexed by the Search Engine.
    /// Folders are NOT recursively searched.
    /// </summary>
    public class SearchDomain
    {
        public string Folder { get; }
        public IList<string> Files { get; }

		/// <summary>
		/// Create a Search Domain of a folder
		/// </summary>
		/// <param name="folder">Path of folder to index</param>
        public SearchDomain(string folder)
        {
            Folder = folder;
            Files = BuildDomain();
        }

        private IList<string> BuildDomain()
        {
            IList<string> files = new List<string>();
            // Step Three: Complete the BuildDomain method



            return files;
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            foreach (string file in Files)
            {
                sb.Append(file);
                sb.Append("\n");
            }
            return sb.ToString();
        }
    }
}
